docs/unit_tests.md

```diff
@@ -0,0 +1,98 @@
+
+### Our First Unit Test
+
+```java
+public class MyUnitTest {
+    // For now we will assume that there is a class called MyUnitTestRule with a test atlas inside called "testAtlas"
+    @Rule
+    public MyUnitTestRule setup = new MyUnitTestRule();
+
+    @Rule
+    public ConsumerBasedExpectedCheckVerifier verifier = new ConsumerBasedExpectedCheckVerifier();
+
+    @Test
+    public void testMyUnitTest()
+    {
+        // MyCheck would be replaced with the name of the check that you are testing
+        // The configuration is optional, and only required if you have specific configuration for the check that you want to test or is required for the check itself
+        this.verifier.actual(this.setup.testAtlas(),
+                new MyCheck(ConfigurationResolver.inlineConfiguration("{\"key\":\"value\"}")));
+        // There are various helper functions described below, this one simply checks if the result of the check on the test atlas produces at least 1 flag.
+        this.verifier.verifyNotEmpty();
+        
+        // Another common scenario would be to loop through the results and check something on it.
+        this.verifier.verify(flag -> {
+            // Check the flag object for some expected values
+        });
+    }   
+}
+```
+
+A basic Atlas-Checks unit test contains 3 things:
+1. A TestRule class that is essentially and atlas or multiple atlas to test against.
+2. A `ConsumerBasedExpectedCheckVerifier` instance that will execute the check you are testing against the test atlas.
+3. A series of units test using the JUnit framework.
+
+### The `ConsumerBasedExpectedCheckVerifier`
+
+The ConsumerBasedExpectedCheckVerifier is a class that will run your check against a test atlas. The test atlas are often very small atlas objects that contain specific data to test against. This verifier has too main types of functions
+
+1. `actual(Atlas, BaseCheck)` - The actual method establishes what data to run against (ie. what atlas file) and what check to run against the provided test atlas.
+2. verify functions - The verify functions will be used to verify the results of the running the check over the test atlas file. Below are some useful functions
+    
+    - `verifyNotEmpty()` - Verifies that at least 1 flag was produced by the check
+    - `verifyEmpty()` - Verifies that no flags were produced by the check
+    - `verifyExpectedSize(int)` - Verifies that a specific number of flags were produced by the check
+    - `verify(Consumer<List<CheckFlag>>)` - A custom verifier allowing the developer to loop through each individual flag that is produced and verify it in any way that the developer wishes.
+    
+ ### The `TestRule`
+ 
+ As mentioned previously the TestRule is a class that is associated with the Unit test class and contains all the test data, which would essentially be an Atlas in some form or another. An Atlas can be created in 2 primary ways:
```

Comment:
There are more now! We can create an Atlas from an OSM file (either from JOSM or downloaded from openstreetmap). Consider adding these options here.

Reply:
There is a task to revamp the atlas documentation around testing, once that's done, we can link it here.

Reply:
So yes I considered adding all the options, however I focused on the 2 that I thought might be the easiest (I might be wrong about that assessment), and then once that documentation is updated we can link directly to it for more detailed information around it.

Reply:
Fair enough. You could argue that an OSM file is easier to create than a text file. Text Atlases are usually the piece of interest from a larger Atlas or if created from scratch, you need to have prior understanding of the text atlas format. OSM files can be downloaded right from OSM or quickly drawn/downloaded in JOSM and saved. Most power users would probably use this over a text atlas. Either one would work, only point was mentioning them at least might help some users.